File: RxJavaRetrofit/app/src/main/java/com/woowrale/rxjavaretrofit/ui/main/MainActivity.java
package com.woowrale.rxjavaretrofit.ui.main;

import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.Toast;
import com.woowrale.rxjavaretrofit.R;
import com.woowrale.rxjavaretrofit.data.model.Crypto;
import com.woowrale.rxjavaretrofit.ui.adapter.RecyclerViewAdapter;
import com.woowrale.rxjavaretrofit.ui.base.BaseActivity;

import java.util.List;

public class MainActivity extends BaseActivity implements MainView {

    RecyclerView recyclerView;
    RecyclerViewAdapter recyclerViewAdapter;
    MainPresenter mPresenter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerViewAdapter = new RecyclerViewAdapter();
        recyclerView.setAdapter(recyclerViewAdapter);

        mPresenter = new MainPresenter();
        mPresenter.setView(this);
        mPresenter.onSearchCryptoCurrency();

    }

    @Override
    public void showMessageError() {
        Toast.makeText(this, "ERROR IN FETCHING API RESPONSE. Try again", Toast.LENGTH_LONG).show();
    }

    @Override
    public void showMessageResult() {
        Toast.makeText(this, "NO RESULTS FOUND", Toast.LENGTH_LONG).show();
    }

    @Override
    public void showListView(List<Crypto.Market> marketList) {
        recyclerViewAdapter.setData(marketList);
    }
}

File: RxDiKotlinRetrofit/app/src/main/java/com/woowrale/rxkotlinretrofit/ui/main/MainActivity.kt
package com.woowrale.rxkotlinretrofit.ui.main

import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.widget.Toast
import com.woowrale.rxkotlinretrofit.R
import com.woowrale.rxkotlinretrofit.data.model.Market
import com.woowrale.rxkotlinretrofit.ui.adapters.RecyclerViewAdapter
import com.woowrale.rxkotlinretrofit.ui.base.BaseActivity

class MainActivity : BaseActivity(), MainView {

    lateinit var recyclerView: RecyclerView
    private lateinit var recyclerViewAdapter: RecyclerViewAdapter
    private lateinit var mPresenter: MainPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mPresenter = MainPresenter()
        mPresenter.setView(this)

        recyclerView = findViewById(R.id.recyclerView)
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerViewAdapter = RecyclerViewAdapter()
        recyclerView.setAdapter(recyclerViewAdapter)

        mPresenter.onCallEndPoint()
    }

    override fun setRecyclerViewAdapterData(marketList: List<Market>) {
        recyclerViewAdapter.setData(marketList)
    }

    override fun showMessageResult() {
        Toast.makeText(this, "NO RESULTS FOUND", Toast.LENGTH_LONG).show()
    }

    override fun showMessageError() {
        Toast.makeText(this, "ERROR IN FETCHING API RESPONSE. Try again", Toast.LENGTH_LONG).show()
    }

}
